### PR TITLE
verify key type in atom.set_value

### PR DESCRIPTION
**Pull request by noskill**
Check that key in atom.set_value is an instance of value

fix for issue https://github.com/opencog/atomspace/issues/2282

### CODE REVIEW

File: opencog/cython/opencog/atom.pyx [hunk 1]
```diff
@@ -62,6 +62,8 @@ cdef class Atom(Value):
             atom_ptr.setTruthValue(deref((<TruthValue>truth_value)._tvptr()))
 
     def set_value(self, key, value):
+        if not isinstance(key, Value):
+            raise RuntimeError("key should be an instance of Value, got {0} instead".format(type(key)))
```
Comment (vsbogd) [inline opencog/cython/opencog/atom.pyx]: I would suggest using `TypeError` as it suits this kind of issue better (see https://docs.python.org/3/library/exceptions.html#TypeError)
```suggestion
            raise TypeError("key should be an instance of Value, got {0} instead".format(type(key)))
```

File: tests/cython/atomspace/test_atom.py [hunk 2]
```diff
@@ -62,6 +62,13 @@ def test_get_input(self):
 
         self.assertEqual(set(incoming), set([a, b]))
 
+    def test_invalid_key(self):
+        string_node = ConceptNode("String")
+        error_str = "key should be an instance of Value, got {0} instead".format(str)
+        with self.assertRaisesRegex(RuntimeError, error_str):
```
Comment (vsbogd) [inline tests/cython/atomspace/test_atom.py]:
```suggestion
        with self.assertRaisesRegex(TypeError, error_str):
```

File: opencog/cython/opencog/atom.pyx [hunk 3]
```diff
@@ -62,6 +62,8 @@ cdef class Atom(Value):
             atom_ptr.setTruthValue(deref((<TruthValue>truth_value)._tvptr()))
 
     def set_value(self, key, value):
+        if not isinstance(key, Value):
+            raise TypeError("key should be an instance of Value, got {0} instead".format(type(key)))
```
Comment (linas) [inline opencog/cython/opencog/atom.pyx]: In the C++ API, the keys are required to be Atoms.

File: tests/cython/atomspace/test_atom.py [hunk 4]
```diff
@@ -62,6 +62,13 @@ def test_get_input(self):
 
         self.assertEqual(set(incoming), set([a, b]))
 
+    def test_invalid_key(self):
+        string_node = ConceptNode("String")
```
Comment (linas) [inline tests/cython/atomspace/test_atom.py]: object created but never used...

Comment (noskill) [reply tests/cython/atomspace/test_atom.py]: string_node used for setting a value

**Pull request merged**